tests/datasource/gnomad/test_gnomad_ld.py

```diff
@@ -65,3 +69,63 @@ def test_resolve_variant_indices(
     assert (
         observed_df.orderBy(observed_df["r"].desc()).collect() == expected_df.collect()
     )
+
+
+@dataclass
+class TestGnomADLDMatrix:
+    """Test GnomAD LD methods."""
+
+    gnomad_ld_matrix = GnomADLDMatrix
+
+    @pytest.fixture(scope="class")
+    def _setup(self, spark: SparkSession) -> None:
+        """Initialize GnomADLDMatrix object."""
+        hl.init(sc=spark.sparkContext, log="/dev/null")
+
+        self.gnomad_ld_matrix = GnomADLDMatrix(
+            ld_matrix_template="tests/data_samples/example_{POP}.bm"
+        )
+
+    def test_ld_matrix_slice(self) -> None:
```

Comment:
Ideally I'd break this logic into different tests. 
What you can do:
1. Add `matrix_slice` to `_setup` so that it can be used as a fixture within the class
```
self.matrix_slice = (
            self.gnomad_ld_matrix
            # Extract a 2 by 2 squary matrix:
            .get_ld_matrix_slice(
                gnomad_ancestry="test-pop", start_index=1, end_index=2
            ).persist()
        )
```
2. Break it into different tests. E.g.:
- Type assertion of `self.matrix_slice`
- Count assertion of `self.matrix_slice`
- Assertion that transversing of `self.matrix_slice` worked 

Just a suggestion if you think it's a good idea. The main point is to make tests as unitary as possible.

Reply:
Thanks, now the single test method was broken down to 4.